database_normalization/database_normalization.md

```diff
@@ -0,0 +1,370 @@
+| subj_id  | street_address  | city  | state   | zip  | date_start  | date_end   |
+| :--------- | :--------- | :--------- | :--------- | :--------- | :--------- | :--------- |
+| 11234   | 123 Main Street   | Smithtown    | PA  | 19000    | 2022-01-01   | `NULL`    |
+| 11234   | 123 Oak Lane   | Old Towne    | PA  | 18000   | 2000-01-01    | 2021-12-31    |
+| 93452   | 123 Green Blvd  | Kirby    | TN  | 37000    | 2020-05-01    | `NULL`   |
+
+What you want to eventually end up with will be a single table of results that might only have three columns: "subj\_id", "dep\_total", and "zip".  That way you can look at the relationship between depression inventory scale and zip code.
+
+You may be asking yourself, "Why don't the people collecting the data or designing the database just put all the information together in one table to start with?"  After all, getting one row per subject (or per sample, etc.) is how we usually prepare to apply statistical tests. If we could just **start** with one row per research subject in our source data, we'd be ready to jump into the science and not have so much data preparation work. 
+
+The reason why this usually doesn't happen in a database, and you instead see lots of different tables that you have to join together is because of a concept known as **normalization**.  To explain normalization, we'll first explain **one-to-many** data relationships.
+
+## One-to-Many
+
+Data often has one-to-many relationships. What does one-to-many mean? Consider two use cases, a hospital and a retail website.  Each organization has one-to-many data relationships:
+
+**Hospital one-to-many examples**
+
+* One patient has multiple encounters on different dates.
+* One encounter may include multiple orders for different procedures and medications.
+* One medication order can give rise to multiple medication administrations.
+
+**Retail website one-to-many examples**
+
+* One customer may have several associated addresses for shipping, billing, or both.
+* One customer will probably have multiple orders.
+* Each order will have one or many associated items.
+
+One-to-many data relationships (one X is related to more than one Y) can be very hard to accommodate in a single table with all the data.  Imagine having a table of order information from an online website, in which a single row contained every item included in that order. 
+
+It would be impossible to know ahead of time how many columns would be needed.  What if we provided enough columns for 100 items in an order to start?  Then, maybe someone orders 110 items.  What do we do?  Also, most orders are much smaller, which means we have many empty cells with no data in them.  It's not an efficient way to store data.  
+
+### Quiz: One-to-Many 
+
+Which of the following relationships are likely to be one-to-many?  Select all the correct answers.
+
+[[X]] Gym member to workout
+[[ ]] Street address to corresponding legislative district
```

Comment:
```suggestion
[[ ]] Street address to latitude and longitude coordinates
```
If you want X:Y to be one to one and X is "street address," the only thing I can think of for Y is its lat/long. Anything regarding a political boundary either has to get way too specific to actually be one-to-one e.g. "state legislative district in a US state with a unicameral legislature" or doesn't work in parts of the world where borders are disputed, e.g. "country."

Reply:
so much better!  agree